docs/markdown/snippets/add_project_dependencies.md

```diff
@@ -0,0 +1,12 @@
+## `add_project_dependencies()` function
+
+Dependencies can now be added to all build products using
+`add_project_dependencies()`.  This can be useful in several
+cases:
+
+* with special dependencies such as `dependency('threads')`
+* with system libraries such as `find_library('m')`, or
+core dependencies such as `glib`
```

Comment:
nitpicking: I don't think glib is a good example here because it will often have subproject fallback and in that case it won't be allowed in `add_project_dependencies()`, unless you do `if glib_dep.type_name() != 'internal'`.

Reply:
Ok, I'll remove it.